docker/package/tezos_baking_wizard.py

```diff
@@ -0,0 +1,489 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+"""
+A wizard utility to help set up tezos-baker.
+
+Asks questions, validates answers, and executes the appropriate steps using the final configuration.
+"""
+
+import os, sys, subprocess, shlex
+import readline
+import re, textwrap
+import urllib.request
+from typing import List
+
+# Input validators
+
+
+def enum_range_validator(options):
+    def _validator(input):
+        intrange = list(map(str, range(1, len(options) + 1)))
+        if input not in intrange and input not in options:
+            raise ValueError(
+                "Please choose one of the provided values or use their respective numbers: "
+                + ", ".join(options)
+            )
+        try:
+            opt = int(input) - 1
+        except:
+            return options[input]
+        else:
+            return list(options.keys())[opt]
+
+    return _validator
+
+
+def filepath_validator(input):
+    if input and not os.path.isfile(input):
+        raise ValueError("Please input a valid file path.")
+    return input
+
+
+def required_field_validator(input):
+    if not input.strip():
+        raise ValueError("Please provide this required option.")
+    return input
+
+
+class Validator:
+    def __init__(self, validator):
+        self.validator = validator
+
+    def validate(self, input):
+        if self.validator is not None:
+            if isinstance(self.validator, list):
+                for v in self.validator:
+                    input = v(input)
+                return input
+            else:
+                return self.validator(input)
+
+
+# Wizard CLI utility
+
+
+def proc_call(cmd):
+    return subprocess.check_call(shlex.split(cmd))
+
+
+def fetch_snapshot(url):
+    print("Downloading the snapshot from ", url)
+    filename = "/tmp/tezos_node.snapshot"
+    urllib.request.urlretrieve(url, filename, progressbar_hook)
+    print()
+    return filename
+
+
+def progressbar_hook(chunk_number, chunk_size, total_size):
+    done = chunk_number * chunk_size
+    percent = min(int(done * 100 / total_size), 100)
+    print("Progress: ", percent, "%, ", int(done / (1024 * 1024)), "MB", end="\r")
+
+
+def pprint_table(options):
```

Comment:
It looks like this is always called before assigning a "table" to the `options` of a `Step`.

Why not move this to be method of the `Step` class, say `print_options` that will:
1. check for `None` and print nothing in that case.
2. use the logic defined here to print the steps otherwise

Additionally I'd suggest to put a `(default)` near the default option to make clear what happens when a user presses Enter.